mlflow/gateway/providers/anthropic.py

```diff
@@ -220,11 +223,29 @@ def __init__(self, config: RouteConfig) -> None:
         if config.model.config is None or not isinstance(config.model.config, AnthropicConfig):
             raise TypeError(f"Invalid config type {config.model.config}")
         self.anthropic_config: AnthropicConfig = config.model.config
-        self.headers = {
+
+    @property
+    def headers(self) -> dict[str, str]:
+        return {
             "x-api-key": self.anthropic_config.anthropic_api_key,
             "anthropic-version": self.anthropic_config.anthropic_version,
         }
-        self.base_url = "https://api.anthropic.com/v1/"
+
+    @property
+    def base_url(self) -> str:
+        return "https://api.anthropic.com/v1/"
```

Comment:
```suggestion
        return "https://api.anthropic.com/v1"
```

Can we remove a trailing slash for consistency?

Reply:
Or add a method for constructing a URL? `f"{self.base_url}messages"` looks error-prone (e.g. `self.base_url` is missing `/`).

```python
def url(self, path: str) -> str:
    return self.base_url.rstrip("/") + path    
```

Reply:
Removing a slash makes sense! (`url` may be a bit redundant to `get_endpoint_url`)